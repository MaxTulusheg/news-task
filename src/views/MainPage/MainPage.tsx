import React from 'react';

import PageLayout from 'components/PageLayout/PageLayout';
import { useNews } from 'hooks/useNews';

import NewsList from './components/NewsList/NewsList';

const MainPage = () => {
  const { news, error, isLoading } = useNews();

  return (
    <PageLayout>
      <NewsList news={news} isLoading={isLoading} error={error} />
    </PageLayout>
  );
};

export default MainPage;
