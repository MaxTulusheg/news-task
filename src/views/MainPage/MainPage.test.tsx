import React from 'react';
import { render } from '@testing-library/react';

import MainPage from './MainPage';

describe('MainPage component', () => {
  it('renders correct structure', () => {
    const { container } = render(<MainPage />);

    expect(container).toMatchSnapshot();
  });
});
