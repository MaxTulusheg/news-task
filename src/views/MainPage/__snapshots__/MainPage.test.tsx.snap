// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component renders correct structure 1`] = `
<div>
  <header
    class="wrapper header"
  >
    <nav
      class="navigation"
    >
      <a
        class="link"
        href="/"
        title="News"
      >
        <img
          alt="News logo"
          class="logo"
          src="logo.svg"
        />
      </a>
    </nav>
  </header>
  <div
    class="wrapper"
  >
    <main
      class="content"
    >
      <section
        class="wrapper"
      >
        <h1
          class="title title h1"
        >
          <a
            href="/"
            title="See all news"
          >
            Most popular News
          </a>
        </h1>
        <div
          class="wrapper loader"
          data-testid="loader"
        >
          <div
            class="loader loader"
          />
        </div>
      </section>
    </main>
    <aside
      class="wrapper aside"
    >
      <h3
        class="title heading h3"
      >
        <img
          alt="Icon"
          class="icon icon"
          src="fire.svg"
        />
        <a
          class="title"
          href="#hot-news"
          title="See all hot news"
        >
          Hot News
        </a>
        <img
          alt="Icon"
          class="icon icon"
          src="fire.svg"
        />
      </h3>
      <div
        class="wrapper loader"
        data-testid="loader"
      >
        <div
          class="loader loader"
        />
      </div>
      <button
        class="refresh"
        disabled=""
        title="Refresh hot news"
      >
        Refresh
        <img
          alt="Icon"
          class="icon refreshIcon refreshLoad"
          src="refresh.svg"
        />
      </button>
    </aside>
  </div>
</div>
`;
